jetty-server/src/main/java/org/eclipse/jetty/server/ResourceService.java

```diff
@@ -401,8 +401,13 @@ protected void sendWelcome(HttpContent content, String pathInContext, boolean en
 
         if (welcome != null)
         {
+            String servletPath = included ? (String)request.getAttribute(RequestDispatcher.INCLUDE_SERVLET_PATH)
+                    : request.getServletPath();
+            if (servletPath == null)
```

Comment:
I think this is a good solution. I would delete the extra "if" at line 406 because that can never happen, unless there's a bug in jetty :)